### ISSUE TITLE
Parte09 ex5 issue

### ISSUE THREAD

**Issue by Nunoc99** [miguelriemoliveira/psr_22-23#11]
I'm trying to solve an error that keeps popping up on ex5, that is that the module I've created it's not being recognized when I try to import the Dog message from "psr_parte09_ex4_bringup", the error message is in the picture below as well as the code.
I've created multiple different packages and tried to run it, and still the same error was shown.
Previously in ex4 from parte08 I had the same issue, but then I made "catkin_make" in the catkin_ws directory and the program ran with no problem, so I tried this on ex5 but it didn't work. I also compared my "CMakeLists.txt" and "package.xml" with the ones made by the professor and they are correct.
I feel like that's some minor detail that that is missing and I can't be able to notice what it is.

![fcf82a15-ed4f-4a38-834e-cff986c49a88](https://user-images.githubusercontent.com/114221939/202863522-c45d08c3-943e-4d32-865f-3b83d6a9a6fe.jpeg)
![Screenshot from 2022-11-19 17-38-58](https://user-images.githubusercontent.com/114221939/202864186-6f9d83a9-05df-4ba7-9ae8-4e9ee4d40317.png)

**Comment by miguelriemoliveira** [comment]
Hi Nuno,

Please post the contents of your .bashrc file.

**Comment by Nunoc99** [comment]
Good morning,

I'm sending a copy of my bashrc file in txt, for some reason here and by e-mail they don't support that type of file.
[bashrc_copy.txt](https://github.com/miguelriemoliveira/psr_22-23/files/10049532/bashrc_copy.txt)

**Comment by miguelriemoliveira** [comment]
Hi @Nunoc99 ,

First about the style.

About supporting txt file, you can just add the content like this 

```
# ~/.bashrc: executed by bash(1) for non-login shells.
# see /usr/share/doc/bash/examples/startup-files (in the package bash-doc)
# for examples

# If not running interactively, don't do anything
case $- in
    *i*) ;;
      *) return;;
esac

# don't put duplicate lines or lines starting with space in the history.
# See bash(1) for more options
HISTCONTROL=ignoreboth
(...)
```

You should also avoid uploading images with code, better like this:

```python
def main():
    print('Hello')
```

**Comment by miguelriemoliveira** [comment]
Now the content:

You have these three lines in your .bashrc

```
source /opt/ros/noetic/setup.bash
source ~/catkin_ws/devel/setup.bash
export ROS_PACKAGE_PATH=~/catkin_ws/src:${ROS_PACKAGE_PATH}
```

So in principle ROS should be properly configured, and you should be able to roscd into your package

Can you do this?

    roscd psr_parte09_ex4_bringup

If that runs ok then the problem should be in the message.

Did you follow the instructions for compiling the message? Did you run catkin_make to compile?

Can you check that the message exists?

    rosmsg show psr_parte09_ex4_bringup/Dog

**Comment by Nunoc99** [comment]
Should I have something more than those three lines?

Yes I can.
Yes I did, I created the message, then I compiled it in the "catkin_ws" directory.
Yes, the message exists.

**Comment by Nunoc99** [comment]
I noticed something that might be the problem, but I don't know how it can be solved. In the catkin_ws, in the folder "build", inside "psr22", I only have the folder of "Parte_08", "Parte_09" isn't there for some reason...
![Screenshot from 2022-11-20 14-40-35](https://user-images.githubusercontent.com/114221939/202908508-1d2fa4df-d16c-4a74-ac5d-8c846f5cb842.png)

**Comment by miguelriemoliveira** [comment]
Perhaps what's happening is that you are trying to use the Dog msg in a different ros package from the one where the message was defined.

Is that the case? If so there extra stuff you need to do in addition to what I've shown in the class.

https://answers.ros.org/question/206257/catkin-use-ros-message-from-another-package/

**Comment by Nunoc99** [comment]
I don't think so, because I created again the Dog msg in the new package.

**Comment by miguelriemoliveira** [comment]
you're right ... this works for you

    rosmsg show psr_parte09_ex4_bringup/Dog

I am not sure what's the problem ... I have to see online.

I should set up a meeting some night this week for everyone who's having problems installing ROS. We can talk there.

**Comment by Nunoc99** [comment]
Yes it works. I've been around this for some time, and I can't figure out what is missing... I'm going to move forward.

Yes, that would be great and thank you anyways.
